lib/aiken/interval.ak

```diff
@@ -355,6 +355,114 @@ pub fn after(lower_bound: a) -> Interval<a> {
   }
 }
 
+// TODO: Replace 'Int' with a generic 'a' once we have comparable traits.
+
+/// Check whether the interval is entirely after the point "a"
+///
+/// ```aiken
+/// is_entirely_after(interval.after(10), 5) == True
+/// is_entirely_after(interval.after(10), 10) == False
+/// is_entirely_after(interval.after(10), 15) == False
+/// is_entirely_after(interval.between(10, 20), 30) == False
+/// is_entirely_after(interval.between(10, 20), 5) == True
+pub fn is_entirely_after(interval: Interval<Int>, point: Int) -> Bool {
```

Comment:
One thing we've been doing to allow things to be a bit more generic is to manually accept an anonymous function that returns the prelude type `Ordering` (might be called `Order`, then internally within the function you can use this function to compare values.

Reply:
isn't this exactly how typeclasses work under the hood in haskell? lol

Reply:
not sure. This is just the convention for now until we do something like Elm with a comparable type or just full blown traits, neither of which are going to happen very soon.

Reply:
want me to make the same change to `contains` as well?

Reply:
Well I'm not necessarily asking for changes, mostly just wanted to share what we've done in other places in stdlib. If this would mostly always be used with Int anyways then this may even be preferable. I'd see what @KtorZ thinks before doing extra work.

Reply:
as a side note, does defining it for Int make it work for PosixTime, which is an alias for Int?

Reply:
yea aliases are not special

Reply:
That's the approach I also took for generic data structures too @rvcas. But, I treated interval differently because I couldn't justify generic intervals in any reasonable use case. The use of that particular interval structure is almost ill-advised to begin with but it's forced by the script context. So I barely see when anyone will want to have an Interval of something else than an Int. 

Hence, to avoid needlessly overloading this module's API, I simply defaulted to Int in cases where comparison are needed and kept other functions generic when possible.

Reply:
Cool I had a feeling that Interval with any thing other than Int was probably not a concern.

Reply:
But what about my totally legitimate use case for `Interval<ScriptContext>`? :( /s